File: tests/dynamic_fixtures/js/sqli_unsupported.js
// SQL injection — unsupported fixture.
// The entry point accepts an HTTP request object, not a plain string.
// HarnessSpec derivation will map this to HttpBody slot (unsupported by JS emitter).
// Entry: handleRequest(req)  Cap: SQL_QUERY
// Expected verdict: Unsupported (EntryKindUnsupported or ConfidenceTooLow)

class UserRepository {
    findUser(req) {
        const name = req && req.body && req.body.name;
        const query = "SELECT * FROM users WHERE name='" + name + "'";
        return query;
    }
}

module.exports = { UserRepository };

File: tests/dynamic_fixtures/rust/sqli_unsupported.rs
/// SQL injection — unsupported entry-kind fixture.
///
/// The vulnerable logic lives inside a struct method. The test creates a Diag
/// with an unsupported entry kind, so `HarnessSpec::from_finding` returns
/// `Err(UnsupportedReason::EntryKindUnsupported)`.
///
/// Expected verdict: Unsupported(EntryKindUnsupported)
/// Cap: SQL_QUERY
pub struct UserRepository;

impl UserRepository {
    pub fn find_user(&self, name: &str) -> Vec<String> {
        use rusqlite::Connection;
        let conn = Connection::open_in_memory().expect("open db");
        let query = format!("SELECT name FROM users WHERE name='{}'", name);
        match conn.prepare(&query) {
            Ok(mut stmt) => stmt
                .query_map([], |row| row.get::<_, String>(0))
                .map(|rows| rows.flatten().collect())
                .unwrap_or_default(),
            Err(_) => vec![],
        }
    }
}
